drivers/pinctrl/pinctrl_mchp_xec.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2016 Open-RnD Sp. z o.o.
+ * Copyright (c) 2021 Linaro Limited
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ * Copyright (c) 2021 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_xec_pinctrl
+
+#include <drivers/pinctrl.h>
+#include <gpio/gpio_mchp_xec.h>
+#include <soc.h>
+
+/*
+ * const struct device *dev = pointer to gpio_xxx_yyy device
+ * uint32_t pin = encoded port and pin number
+ * uint32_t cfg = property encodings: pulls, buffer type, slew, drive strength
+ * uint32_t altf = function number, 0 = GPIO, > 0 alternate function
+ */
+static int mchp_xec_pin_configure(const struct device *dev, uint32_t portpin,
+				  uint32_t cfg, uint32_t altf)
+{
+	uint32_t pin = MCHP_XEC_PINMUX_PIN(portpin);
+
+	if (!dev || !device_is_ready(dev)) {
+		return -ENODEV;
+	}
+
+	return gpio_mchp_xec_configure(dev, pin, cfg, altf);
+}
+
+int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt,
+			   uintptr_t reg)
+{
+	const struct device *dev;
+	uint32_t portpin, mux, cfg, func;
+	int ret;
+
+	ARG_UNUSED(reg);
+
+	for (uint8_t i = 0U; i < pin_cnt; i++) {
+		mux = pins[i].pinmux;
+
+		func = MCHP_XEC_PINMUX_FUNC(mux);
+		if (func >= MCHP_AFMAX) {
+			return -EINVAL;
+		}
+
+		cfg = pins[i].pincfg;
+		portpin = MEC_XEC_PINMUX_PORT_PIN(mux);
+
+		dev = (const struct device *)pins[i].gpioph;
```

Comment:
I feel this is mixing GPIO/pinctrl stuff really. Other platforms such as STM32/GD32 where pinmux is done using GPIO registers don't do anything like this. I'd prefer to stick with the initial approach.

Reply:
The phandle was introduced to address Maureen's [comment](https://github.com/zephyrproject-rtos/zephyr/pull/41888#discussion_r791047583). 

Can you confirm you are okay with not using phandle and using the `gpio_ports[]` table as before?

Reply:
Another approach (as suggested in the [comment](https://github.com/zephyrproject-rtos/zephyr/pull/41888#discussion_r791047583)) is to resolve this issue  as a separate PR. 

We would like to proceed with the phandle approach for now.

Reply:
Thanks for confirming; will revert to previous solution as described in the [comment](https://github.com/zephyrproject-rtos/zephyr/pull/41888#discussion_r809409996).